We have race in logs due to this PR, will fix:
```
2026-04-28T18:03:28.7341680Z === NAME  TestRPC/websocket
2026-04-28T18:03:28.7342338Z     logger.go:146: 2026-04-28T18:01:33.321Z	INFO	websocket client disconnected	{"remoteAddr": "127.0.0.1:45264", "numOfSubs": 0}
2026-04-28T18:03:28.7343175Z     logger.go:146: 2026-04-28T18:01:33.330Z	INFO	websocket client connected	{"remoteAddr": "127.0.0.1:45274", "numOfSubs": 1}
2026-04-28T18:03:28.7344052Z     logger.go:146: 2026-04-28T18:01:33.351Z	DEBUG	processing rpc request	{"method": "invokecontainedscript", "params": "[  Application ]"}
2026-04-28T18:03:28.7345378Z     logger.go:146: 2026-04-28T18:01:33.352Z	INFO	Error encountered with rpc request	{"code": -32602, "cause": "invalid verbose parameter: not a boolean", "method": "invokecontainedscript", "params": "[  Application ]"}
2026-04-28T18:03:28.7346180Z === RUN   TestRPC/websocket/invokecontainedscript/good,_tx
2026-04-28T18:03:28.7346490Z === NAME  TestRPC/websocket
2026-04-28T18:03:28.7347083Z     logger.go:146: 2026-04-28T18:01:33.371Z	INFO	websocket client disconnected	{"remoteAddr": "127.0.0.1:45274", "numOfSubs": 0}
2026-04-28T18:03:28.7347977Z     logger.go:146: 2026-04-28T18:01:33.381Z	INFO	websocket client connected	{"remoteAddr": "127.0.0.1:45284", "numOfSubs": 1}
2026-04-28T18:03:28.7348790Z     logger.go:146: 2026-04-28T18:01:33.401Z	DEBUG	processing rpc request	{"method": "invokecontainedscript", "params": "[]"}
2026-04-28T18:03:28.7349353Z === RUN   TestRPC/websocket/invokecontainedscript/good,_tx_and_block
2026-04-28T18:03:28.7349688Z === NAME  TestRPC/websocket
2026-04-28T18:03:28.7350258Z     logger.go:146: 2026-04-28T18:01:33.441Z	INFO	websocket client disconnected	{"remoteAddr": "127.0.0.1:45284", "numOfSubs": 0}
2026-04-28T18:03:28.7351068Z     logger.go:146: 2026-04-28T18:01:33.442Z	INFO	websocket client connected	{"remoteAddr": "127.0.0.1:45292", "numOfSubs": 1}
2026-04-28T18:03:28.7351877Z     logger.go:146: 2026-04-28T18:01:33.462Z	DEBUG	processing rpc request	{"method": "invokecontainedscript", "params": "[ ]"}
2026-04-28T18:03:28.7352425Z === RUN   TestRPC/websocket/invokecontainedscript/good,_all_args
2026-04-28T18:03:28.7352751Z === NAME  TestRPC/websocket
2026-04-28T18:03:28.7353315Z     logger.go:146: 2026-04-28T18:01:33.483Z	INFO	websocket client disconnected	{"remoteAddr": "127.0.0.1:45292", "numOfSubs": 0}
2026-04-28T18:03:28.7354185Z     logger.go:146: 2026-04-28T18:01:33.512Z	INFO	websocket client connected	{"remoteAddr": "127.0.0.1:45300", "numOfSubs": 1}
2026-04-28T18:03:28.7355248Z     logger.go:146: 2026-04-28T18:01:33.532Z	DEBUG	processing rpc request	{"method": "invokecontainedscript", "params": "[  Application]"}
2026-04-28T18:03:28.7356046Z     logger.go:146: 2026-04-28T18:01:33.550Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:43655"}
2026-04-28T18:03:28.7356444Z ==================
2026-04-28T18:03:28.7356644Z WARNING: DATA RACE
2026-04-28T18:03:28.7356878Z Read at 0x00c002b3ee4b by goroutine 15458:
2026-04-28T18:03:28.7357180Z   testing.(*common).destination()
2026-04-28T18:03:28.7357639Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1049 +0x96
2026-04-28T18:03:28.7358013Z   testing.(*common).log()
2026-04-28T18:03:28.7358431Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1027 +0xbe
2026-04-28T18:03:28.7358778Z   testing.(*common).Logf()
2026-04-28T18:03:28.7359189Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1191 +0x8e
2026-04-28T18:03:28.7359596Z   go.uber.org/zap/zaptest.TestingWriter.Write()
2026-04-28T18:03:28.7360126Z       /home/runner/go/pkg/mod/go.uber.org/zap@v1.27.1/zaptest/logger.go:146 +0x119
2026-04-28T18:03:28.7360571Z   go.uber.org/zap/zaptest.(*TestingWriter).Write()
2026-04-28T18:03:28.7360898Z       <autogenerated>:1 +0x74
2026-04-28T18:03:28.7361176Z   go.uber.org/zap/zapcore.(*ioCore).Write()
2026-04-28T18:03:28.7361654Z       /home/runner/go/pkg/mod/go.uber.org/zap@v1.27.1/zapcore/core.go:99 +0x18d
2026-04-28T18:03:28.7362069Z   go.uber.org/zap/zapcore.(*CheckedEntry).Write()
2026-04-28T18:03:28.7362565Z       /home/runner/go/pkg/mod/go.uber.org/zap@v1.27.1/zapcore/entry.go:258 +0x1ec
2026-04-28T18:03:28.7362948Z   go.uber.org/zap.(*Logger).Info()
2026-04-28T18:03:28.7363461Z       /home/runner/go/pkg/mod/go.uber.org/zap@v1.27.1/logger.go:247 +0x64
2026-04-28T18:03:28.7363973Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).handleHTTPRequest()
2026-04-28T18:03:28.7364564Z       /home/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:542 +0x10c4
2026-04-28T18:03:28.7365191Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).handleHTTPRequest-fm()
2026-04-28T18:03:28.7365616Z       <autogenerated>:1 +0x51
2026-04-28T18:03:28.7365883Z   net/http.HandlerFunc.ServeHTTP()
2026-04-28T18:03:28.7366332Z       /opt/hostedtoolcache/go/1.25.9/x64/src/net/http/server.go:2322 +0x47
2026-04-28T18:03:28.7366712Z   net/http.serverHandler.ServeHTTP()
2026-04-28T18:03:28.7367170Z       /opt/hostedtoolcache/go/1.25.9/x64/src/net/http/server.go:3340 +0x2a1
2026-04-28T18:03:28.7367531Z   net/http.(*conn).serve()
2026-04-28T18:03:28.7367949Z       /opt/hostedtoolcache/go/1.25.9/x64/src/net/http/server.go:2109 +0xda4
2026-04-28T18:03:28.7368406Z   net/http.(*Server).Serve.gowrap3()
2026-04-28T18:03:28.7368853Z       /opt/hostedtoolcache/go/1.25.9/x64/src/net/http/server.go:3493 +0x4f
2026-04-28T18:03:28.7369107Z 
2026-04-28T18:03:28.7369255Z Previous write at 0x00c002b3ee4b by goroutine 11977:
2026-04-28T18:03:28.7369569Z   testing.tRunner.func1()
2026-04-28T18:03:28.7369995Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1921 +0x904
2026-04-28T18:03:28.7370358Z   runtime.deferreturn()
2026-04-28T18:03:28.7370763Z       /opt/hostedtoolcache/go/1.25.9/x64/src/runtime/panic.go:589 +0x5d
2026-04-28T18:03:28.7371123Z   testing.(*T).Run.gowrap1()
2026-04-28T18:03:28.7371551Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1997 +0x44
2026-04-28T18:03:28.7371802Z 
2026-04-28T18:03:28.7371906Z Goroutine 15458 (running) created at:
2026-04-28T18:03:28.7372174Z   net/http.(*Server).Serve()
2026-04-28T18:03:28.7372593Z       /opt/hostedtoolcache/go/1.25.9/x64/src/net/http/server.go:3493 +0x889
2026-04-28T18:03:28.7372983Z   net/http/httptest.(*Server).goServe.func1()
2026-04-28T18:03:28.7373485Z       /opt/hostedtoolcache/go/1.25.9/x64/src/net/http/httptest/server.go:311 +0xb2
2026-04-28T18:03:28.7373768Z 
2026-04-28T18:03:28.7373990Z Goroutine 11977 (finished) created at:
2026-04-28T18:03:28.7374333Z   testing.(*T).Run()
2026-04-28T18:03:28.7374740Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1997 +0x9d2
2026-04-28T18:03:28.7375285Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.TestRPC()
2026-04-28T18:03:28.7375856Z       /home/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server_test.go:2392 +0x56
2026-04-28T18:03:28.7376230Z   testing.tRunner()
2026-04-28T18:03:28.7376644Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1934 +0x21c
2026-04-28T18:03:28.7377006Z   testing.(*T).Run.gowrap1()
2026-04-28T18:03:28.7377429Z       /opt/hostedtoolcache/go/1.25.9/x64/src/testing/testing.go:1997 +0x44
2026-04-28T18:03:28.7377753Z ==================
2026-04-28T18:03:28.7377950Z === NAME  TestRPC
2026-04-28T18:03:28.7378539Z     logger.go:146: 2026-04-28T18:01:33.564Z	INFO	websocket client disconnected	{"remoteAddr": "127.0.0.1:45300", "numOfSubs": 0}
2026-04-28T18:03:28.7379148Z     testing.go:1617: race detected during execution of test
2026-04-28T18:03:28.7379455Z --- FAIL: TestRPC (71.13s)
```

--- a/pkg/core/prometheus.go
+++ b/pkg/core/prometheus.go
@@ -9,16 +9,16 @@ var (
 	// blockHeight prometheus metric.
 	blockHeight = prometheus.NewGauge(
 		prometheus.GaugeOpts{
-			Help:      "Current index of processed block",
-			Name:      "current_block_height",
+			Help:      "Index of processed block",
+			Name:      "block_height",
 			Namespace: "neogo",
 		},
 	)
 	// persistedHeight prometheus metric.
 	persistedHeight = prometheus.NewGauge(
 		prometheus.GaugeOpts{
-			Help:      "Current persisted block count",
-			Name:      "current_persisted_height",
+			Help:      "Persisted block count",
+			Name:      "persisted_height",
 			Namespace: "neogo",
 		},
 	)
@@ -33,8 +33,8 @@ var (
 	// headerHeight prometheus metric.
 	headerHeight = prometheus.NewGauge(
 		prometheus.GaugeOpts{
-			Help:      "Current header height",
-			Name:      "current_header_height",
+			Help:      "Header height",
+			Name:      "header_height",
 			Namespace: "neogo",
 		},
 	)

--- a/pkg/core/stateroot/prometheus.go
+++ b/pkg/core/stateroot/prometheus.go
@@ -5,8 +5,8 @@ import "github.com/prometheus/client_golang/prometheus"
 // stateHeight prometheus metric.
 var stateHeight = prometheus.NewGauge(
 	prometheus.GaugeOpts{
-		Help:      "Current verified state height",
-		Name:      "current_state_height",
+		Help:      "Verified state height",
+		Name:      "verified_state_height",
 		Namespace: "neogo",
 	},
 )

--- a/pkg/services/rpcsrv/prometheus.go
+++ b/pkg/services/rpcsrv/prometheus.go
@@ -9,7 +9,14 @@ import (
 
 // Metrics used in monitoring service.
 var (
-	rpcTimes = map[string]prometheus.Histogram{}
+	rpcTimes         = map[string]prometheus.Histogram{}
+	wsConnectionsCnt = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Help:      "WS connections count (local and remote clients)",
+			Name:      "wsconnections_count",
+			Namespace: "neogo",
+		},
+	)
 )
 
 func addReqTimeMetric(name string, t time.Duration) {
@@ -37,4 +44,9 @@ func init() {
 	for call := range rpcWsHandlers {
 		regCounter(call)
 	}
+	prometheus.MustRegister(wsConnectionsCnt)
+}
+
+func updateWSConnectionsCnt(cnt int) {
+	wsConnectionsCnt.Set(float64(cnt))
 }

--- a/pkg/services/rpcsrv/server.go
+++ b/pkg/services/rpcsrv/server.go
@@ -533,11 +533,12 @@ func (s *Server) handleHTTPRequest(w http.ResponseWriter, httpRequest *http.Requ
 		subscr := &subscriber{writer: subChan, feeds: make([]feed, s.config.MaxWebSocketFeeds)}
 		s.subsLock.Lock()
 		s.subscribers[subscr] = true
+		numOfSubs = len(s.subscribers)
 		s.subsLock.Unlock()
-		s.log.Info("websocket client connected", zap.String("remoteAddr", remote))
+		updateWSConnectionsCnt(numOfSubs)
+		s.log.Info("websocket client connected", zap.String("remoteAddr", remote), zap.Int("numOfSubs", numOfSubs))
 		go s.handleWsWrites(ws, resChan, subChan)
-		s.handleWsReads(ws, resChan, subscr)
-		s.log.Info("websocket client disconnected", zap.String("remoteAddr", remote))
+		s.handleWsReads(ws, resChan, subscr, remote)
 		return
 	}
 
@@ -574,8 +575,10 @@ func (s *Server) RegisterLocal(ctx context.Context, events chan<- neorpc.Notific
 	subscr := &subscriber{writer: subChan, feeds: make([]feed, s.config.MaxWebSocketFeeds)}
 	s.subsLock.Lock()
 	s.subscribers[subscr] = true
+	numOfSubs := len(s.subscribers)
 	s.subsLock.Unlock()
-	s.log.Info("local websocket client connected")
+	updateWSConnectionsCnt(numOfSubs)
+	s.log.Info("local websocket client connected", zap.Int("numOfSubs", numOfSubs))
 	go s.handleLocalNotifications(ctx, events, subChan, subscr)
 	return func(req *neorpc.Request) (*neorpc.Response, error) {
 		return s.handleInternal(req, subscr)
@@ -682,7 +685,7 @@ eventloop:
 		}
 	}
 	close(events)
-	s.dropSubscriber(subscr)
+	numOfSubs := s.dropSubscriber(subscr)
 drainloop:
 	for {
 		select {
@@ -691,7 +694,7 @@ drainloop:
 			break drainloop
 		}
 	}
-	s.log.Info("local websocket client disconnected")
+	s.log.Info("local websocket client disconnected", zap.Int("numOfSubs", numOfSubs))
 }
 
 func (s *Server) handleWsWrites(ws *websocket.Conn, resChan <-chan abstractResult, subChan <-chan intEvent) {
@@ -746,7 +749,7 @@ drainloop:
 	}
 }
 
-func (s *Server) handleWsReads(ws *websocket.Conn, resChan chan<- abstractResult, subscr *subscriber) {
+func (s *Server) handleWsReads(ws *websocket.Conn, resChan chan<- abstractResult, subscr *subscriber, remoteAddr string) {
 	ws.SetReadLimit(s.wsReadLimit)
 	err := ws.SetReadDeadline(time.Now().Add(wsPongLimit))
 	ws.SetPongHandler(func(string) error { return ws.SetReadDeadline(time.Now().Add(wsPongLimit)) })
@@ -767,15 +770,18 @@ requestloop:
 		case resChan <- res:
 		}
 	}
-	s.dropSubscriber(subscr)
+	numOfSubs := s.dropSubscriber(subscr)
+	s.log.Info("websocket client disconnected", zap.String("remoteAddr", remoteAddr), zap.Int("numOfSubs", numOfSubs))
 	close(resChan)
 	ws.Close()
 }
 
-func (s *Server) dropSubscriber(subscr *subscriber) {
+func (s *Server) dropSubscriber(subscr *subscriber) int {
 	s.subsLock.Lock()
 	delete(s.subscribers, subscr)
+	numOfSubs := len(s.subscribers)
 	s.subsLock.Unlock()
+	updateWSConnectionsCnt(numOfSubs)
 	s.subsCounterLock.Lock()
 	for _, e := range subscr.feeds {
 		if e.event != neorpc.InvalidEventID {
@@ -783,6 +789,7 @@ func (s *Server) dropSubscriber(subscr *subscriber) {
 		}
 	}
 	s.subsCounterLock.Unlock()
+	return numOfSubs
 }
 
 func (s *Server) getBestBlockHash(_ params.Params) (any, *neorpc.Error) {

